### ISSUE TITLE
No longer works since new driver

### ISSUE THREAD

**Issue by thepusher90** [Eliovp/amdmemorytweak#67]
As of the new Update 21.12.1 this tool no longer works with my Vega 56. I will attach the error as screenshot.
![rHWk4aYsUT](https://user-images.githubusercontent.com/37190110/144644727-c1159efb-e22a-4102-acfb-d48fe91e6721.png)
.

**Comment by SystemOO7** [comment]
i got the same problem, i hope there is a fix, because this driver fixes certain issues and i need to use it

**Comment by OverElfstone** [comment]
Got the same problem.
But looks like this project is dead...
Hopefully some one with the knowhow will fork and make a solution.

**Comment by MehlstaubtheCat** [comment]
Hi Guys!

For me its also very important that this tool will work again!
Can someone of you write personally to the guys who made the tool, does any of you know how?

Thx!

**Comment by perestoronin** [comment]
Bugs in new some new kernel drivers, but with latest driver from kernel 5.15.10 work fine again
```
# amdmemorytweak --c
GPU 0:  Vega 10 XTX [Radeon Vega Frontier Edition]	pci:0000:0c:00.0
Memory state: 1000MHz
Memory: Samsung HBM2
Timing 1:	  CL: 17	  RAS: 33	  RCDRD: 16	  RCDWR: 10
Timing 2:	  RCAb (RC): 47	  RCPb (RC): 47	  RPAb (RP): 14	  RPPb (RP): 14
Timing 3:	  RRDS: 4	  RRDL: 6	  RTP: 5
Timing 4:	  FAW: 16
Timing 5:	  CWL: 6	  WTRS: 4	  WTRL: 9
Timing 6:	  WR: 16
Timing 7:	  RREFD: 8
Timing 8:	  RDRDDD: 3	  RDRDSD: 2	  RDRDSC: 1	  RDRDSCL: 3
Timing 9:	  WRWRDD: 3	  WRWRSD: 2	  WRWRSC: 1	  WRWRSCL: 3
Timing 10:	  WRRD: 1	  RDWR: 17
Timing 12:	  REF: 3900
Timing 13:	  MRD: 8	  MOD: 15
Timing 14:	  XS: 270
Timing 16:	  XSMRS: 8
Timing 17:	  PD: 8  	  CKSRE: 10	  CKSRX: 10
Timing 20:	  RFCPB: 160	  STAG: 7
Timing 21:	  XP: 8  	  CPDED: 2	  CKE: 8
Timing 22:	  RDDATA: 16	  WRLAT: 4	  RDLAT: 14	  WRDATA: 1
Timing 23:	  CKESTAG: 2
RFC Timing:	  RFC: 350

# uname -a
Linux ws 5.15.10-gentoo-rt24 #1 SMP PREEMPT_RT Mon Dec 20 03:31:18 MSK 2021 x86_64 AMD Ryzen 9 3900XT 12-Core Processor AuthenticAMD GNU/Linux
```

**Comment by MehlstaubtheCat** [comment]
I use Windows 10 and my OC is just for gaming! 
So i have no other chance, i have to use the newest drivers from AMD to get the best perfomance out of the game right now and in the furture! 
I really hope there are some ppl to fix the AMD memory tweak for the next 2-3 year! 
I will work this long with my Vega 56!
Plz get this awsome tool fixt! Plz!

**Comment by cmcosta81** [comment]
Same issue as OP. I've downgraded to the last working driver for me, 21.11.3. Wish there would be an update or fork, as there have been several new drivers released since then.

**Comment by adrianobellia** [comment]
If you help me to understand how i can find the new function for read and write the timings values i could try to fork it for the new drivers

**Comment by perestoronin** [comment]
Try self-compile snapshot with commit = 9a64ff10eeb449f7f8d286cdc849185269e1c00b

Work fine in linux with kernel 5.16.15

**Comment by adrianobellia** [comment]
@perestoronin i will try but i think something changed in windows drivers and need to fix the code

**Comment by adrianobellia** [comment]
@perestoronin i tried it work! Thanks!!

**Comment by cmcosta81** [comment]
> @perestoronin i tried it work! Thanks!!

You made a build that works with the latest windows drivers?

**Comment by perestoronin** [comment]
> > @perestoronin i tried it work! Thanks!!
> 
> You made a build that works with the latest windows drivers?

under windows I use cards from nvidia, and under linux - from amd

**Comment by adrianobellia** [comment]
@cmcosta81 yes the build linked by @perestoronin 
, i writed yesterday a lil software for link the working memory tweak exe and setting the timings with a graphic interface it stay in background for re-tweaking the ram after windows suspansion, and i put an option for the autoload on windows start, i will fix some bug and i will publicate the source on github i didi it very fast, not the best coding XD

**Comment by jokubasver** [comment]
The CLI version still works fine - it's the GUI version that's not working. I guess ADL2 got updated on recent drivers, and since the GUI version can change clocks, voltages, and fan speed using the ADL2 library, that's why it doesn't work.

As a workaround (and actually my new preferred method now), I simply use the CLI version of amdmemorytweak, together with OverdriveNTool 0.2.9 (using this you can set everything the same as on the GUI version of amdmemorytweak except the timings, save everything to a profile and run it via a command line/script)

So basically I made a batch script that does everything for me:

> C:\amdmemorytweak-master\win\WinAMDTweak (--YOUR TIMINGS HERE)
> C:\OverdriveN\OverdriveNTool (-YOUR GPU AND PROFILE NAME)
> C:\teamredminer-v0.9.4.2-win\teamredminer (-YOUR MINING CONFIG)

(Make sure to run as the script as admin)

**Comment by adrianobellia** [comment]
https://github.com/adrianobellia/Vega-memory-Tweak-Autoloader
